### ISSUE TITLE
Athena dialect do not honor reserved words

### ISSUE THREAD

**Issue by KulykDmytro** [sqlfluff/sqlfluff#3968]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Many of keywords not mentioned in [Athena dialect](https://docs.aws.amazon.com/athena/latest/ug/reserved-words.html) currently treated as reserved by `L029` rule and causing validation issue

### Expected Behaviour

keywords not mentioned [here](https://docs.aws.amazon.com/athena/latest/ug/reserved-words.html) should fail `L029` validation when used f.e. as column name

### Observed Behaviour

```verilog
L: 1 | P:  8 | L029 | Keywords should not be used as identifiers.
```

### How to reproduce

```sql
select domain
from t1
```

### Dialect

athena

### Version

1.3.2

### Configuration

n/a

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by KulykDmytro** [comment]
partially fixed with #3551, #4391, #4490

**Comment by KulykDmytro** [comment]
same on `2.3.2`

**Comment by KulykDmytro** [comment]
same on `3.2.0`
